### PR TITLE
Fix Platinum Blissey_22

### PR DESCRIPTION
**Pull request by luongthomas**
Fix crash on Nurse call ability

### CODE REVIEW

**Comment by luongthomas** [comment]
[BLISSEY_22](https://trello.com/c/Vcy6BlhT/992-blissey22)

File: src/tcgwars/logic/impl/gen4/Platinum.groovy [hunk 1]
```diff
@@ -1009,11 +1009,11 @@ public enum Platinum implements LogicCardInfo {
               checkLastTurn()
               checkNoSPC()
               assert my.hand : "Your hand is empty"
-              assert my.all.find{it.numberOfDamageCounters} : "Your Pokémon are healthy"
+              assert my.all.find {it.numberOfDamageCounters} : "Your Pokémon are healthy"
```
Comment (starg09) [inline src/tcgwars/logic/impl/gen4/Platinum.groovy:1012]: > Q. Can I use Blissey's "Nurse Call" Poke-POWER to discard a card from my hand even if I have no damaged Pokemon in play?
>
> A. Yes, you can. (Jun 11, 2009 PUI Rules Team)

This shouldn't be asserted at all (heal below could be replaced with a filer, then doing the heal only if valid targets are found.

Comment (starg09) [inline src/tcgwars/logic/impl/gen4/Platinum.groovy:1012]: > Q. Can I use Blissey's "Nurse Call" Poke-POWER to discard a card from my hand even if I have no damaged Pokemon in play?
>
> A. Yes, you can. (Jun 11, 2009 PUI Rules Team)

This shouldn't be asserted at all (heal below could be replaced with a filer, then doing the heal only if valid targets are found.

**Review by starg09 — CHANGES_REQUESTED** [review]

File: src/tcgwars/logic/impl/gen4/Platinum.groovy (hunk 1 above)
Comment (starg09) [reply src/tcgwars/logic/impl/gen4/Platinum.groovy]: Not going to lie, I've no clue what I meant to say when I wrote "could be replaced with a filer" 😓 

But with no assert, there's a need for one more change, suggesting it in a second.

File: src/tcgwars/logic/impl/gen4/Platinum.groovy [hunk 2]
```diff
@@ -1009,11 +1009,10 @@ public enum Platinum implements LogicCardInfo {
               checkLastTurn()
               checkNoSPC()
               assert my.hand : "Your hand is empty"
-              assert my.all.find{it.numberOfDamageCounters} : "Your Pokémon are healthy"
               assert bg.em().retrieveObject("Nurse_Call") != bg.turnCount : "You cannot use Nurse Call more than once per turn"
               powerUsed()
               my.hand.select("Choose a card to discard").discard()
-              heal 20, my.all.findAll(it.numberOfDamageCounters).select("Heal which Pokémon"), Source.POKEPOWER
+              heal 20, my.all.findAll{it.numberOfDamageCounters}.select("Heal which Pokémon"), Source.POKEPOWER
```
Comment (starg09) [inline src/tcgwars/logic/impl/gen4/Platinum.groovy:1015]:
```suggestion
              if (my.all.any{it.numberOfDamageCounters}) 
                heal 20, my.all.findAll{it.numberOfDamageCounters}.select("Heal which Pokémon"), Source.POKEPOWER
```

Comment (starg09) [reply src/tcgwars/logic/impl/gen4/Platinum.groovy:1015]: Just to avoid a popup when that part can't be done.

**Review by starg09 — APPROVED** [review]

**Pull request merged**